### PR TITLE
[DRAFT] Implement projects loading and experiments loading for api v2

### PR DESCRIPTION
**Pull request by cosa65**
# Description

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-1803

#### Links to any PRs or resources related to this PR
https://github.com/hms-dbmi-cellenics/api/pull/327

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: src/config/index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { api } from 'utils/constants';
 
 const config = {
-  currentApiVersion: api.V1,
+  currentApiVersion: api.V2,
```
Comment (aerlaut) [inline src/config/index.js]: Don't forget to remove this before merging

File: src/redux/actions/experiments/reorderSamples.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      type: EXPERIMENTS_UPDATED,
+      payload: {
+        experimentId,
+        experiment: { sampleIds: newSampleOrder },
+      },
+    });
+
+    dispatch({
+      type: EXPERIMENTS_SAVED,
+    });
```
Comment (aerlaut) [inline src/redux/actions/experiments/reorderSamples.js]: Can the changes done by the reducer listening to EXPERIMENTS_SAVED be done in EXPERIMENTS_UPDATED? Are they always hand in hand? WE might be able to do some simplification here.

File: src/redux/actions/experiments/updateExperiment.js [hunk 3]
```diff
@@ -47,38 +52,57 @@ const updateExperiment = (
...
+      type: EXPERIMENTS_UPDATED,
+      payload: {
+        experimentId,
+        experiment: experimentDiff,
+      },
+    });
+
+    dispatch({
+      type: EXPERIMENTS_SAVED,
+    });
```
Comment (aerlaut) [inline src/redux/actions/experiments/updateExperiment.js]: Same question as above, can we do these updates together

File: src/redux/actions/experiments/updateExperiment.js [hunk 4]
```diff
@@ -47,38 +52,57 @@ const updateExperiment = (
...
-      });
-    } catch (e) {
-      let { message } = e;
-      if (!isServerError(e)) {
-        console.error(`fetch ${url} error ${message}`);
-        message = endUserMessages.CONNECTION_ERROR;
+    } else if (config.currentApiVersion === api.V2) {
+      if (experimentDiff.sampleIds) {
+        throw new Error('SampleIds in v2 shouldn\'t be updated in this action creator');
       }
```
Comment (aerlaut) [inline src/redux/actions/experiments/updateExperiment.js]: Can you explain what is your intention behind these lines? Do you mean that this should throw an error incase someone is trying to use the action to update sampleIds?

File: src/redux/actions/projects/updateProject.js [hunk 5]
```diff
@@ -22,7 +25,11 @@ const updateProject = (
   const newProject = mergeObjectWithArrays(currentProject, diff);
 
   try {
-    await dispatch(saveProject(projectUuid, newProject));
+    if (config.currentApiVersion === api.V1) {
+      await dispatch(saveProject(projectUuid, newProject));
+    } else if (config.currentApiVersion === api.V2) {
+      // Dont do any fetch, updating the experiment is enough
+    }
```
Comment (aerlaut) [inline src/redux/actions/projects/updateProject.js]: Do we need this `else if` block? Can't the comment be put before the `if` statement above explaning why it'so only done for V1?

File: src/__test__/components/data-management/SamplesTable.test.jsx [hunk 6]
```diff
@@ -207,4 +223,29 @@ describe('Samples table', () => {
       expect(screen.getByText(sampleName)).toBeInTheDocument();
     });
   });
+
+  it('Reorder samples send correct request in api v2', async () => {
```
Comment (aerlaut) [inline src/__test__/components/data-management/SamplesTable.test.jsx]: Does it send the correct request in API v1?

File: src/redux/actions/experiments/updateExperiment.js (hunk 4 above)
Comment (cosa65) [reply src/redux/actions/experiments/updateExperiment.js]: Exactly, updating the `sampleIds` array directly is not possible in any way in the api, this is just a way of making it more clear for developers in the UI side until we can perform the refactor in the UI.

You can either update the samples order through the `/order` endpoint or by hitting the (future) endpoints to add and remove a sample.

File: src/redux/actions/projects/updateProject.js (hunk 5 above)
Comment (cosa65) [reply src/redux/actions/projects/updateProject.js]: ok

**Pull request closed without merging**